packages/curve-ui-kit/src/shared/ui/Balance.tsx

```diff
@@ -3,25 +3,19 @@ import AccountBalanceWalletOutlinedIcon from '@mui/icons-material/AccountBalance
 import Button from '@mui/material/Button'
 import Stack from '@mui/material/Stack'
 import Typography from '@mui/material/Typography'
+import { t } from '@ui-kit/lib/i18n'
 import { WithSkeleton } from '@ui-kit/shared/ui/WithSkeleton'
 import { SizesAndSpaces } from '@ui-kit/themes/design/1_sizes_spaces'
-import type { SxProps } from '@ui-kit/utils'
+import { formatNumber as sharedFormat, SxProps } from '@ui-kit/utils'
 
 const { Spacing, IconSize } = SizesAndSpaces
 
-const formatNumber = (value?: number): string => {
-  if (value == null) return '?'
-  if (value === 0) return '0'
-
-  return value.toLocaleString(undefined, {
-    minimumFractionDigits: 2,
-    maximumFractionDigits: 2,
-  })
-}
+const formatNumber = (value?: number): string =>
+  value == null ? '?' : sharedFormat(value, { decimals: 2, abbreviate: true })
```

Comment:
@0xAlunara is this correct? It didn't seem appropriate to use a custom format

Reply:
why does the new `formatNumber` not handle `null` and `undefined` btw? I am now forced to repeat the `'-'` value which can lead to inconsistency

Reply:
https://github.com/curvefi/curve-frontend/pull/1335